bot/exts/info/subscribe.py

```diff
@@ -133,15 +151,50 @@ async def update_view(self, interaction: Interaction) -> None:
         self.style = self.REMOVE_STYLE if self.assigned else self.ADD_STYLE
         self.label = self.LABEL_FORMAT.format(action="Remove" if self.assigned else "Add", role_name=self.role.name)
         try:
-            await interaction.message.edit(view=self.view)
+            await self.view.anchor_message.edit(view=self.view)
         except discord.NotFound:
             log.debug("Subscribe message for %s removed before buttons could be updated", interaction.user)
             self.view.stop()
 
 
+class AllSelfAssignableRolesView(discord.ui.View):
+    """A persistent view that'll hold one button allowing interactors to toggle all available self-assignable roles."""
+
+    def __init__(self):
+        super(AllSelfAssignableRolesView, self).__init__(timeout=None)
+
+
+class ShowAllSelfAssignableRolesButton(discord.ui.Button):
+    """A button that, when clicked, sends a view a user can interact with to assign one or all the available roles."""
+
+    CUSTOM_ID = "gotta-claim-them-all"
+
+    def __init__(self, assignable_roles: list[AssignableRole]):
+        super().__init__(
+            style=discord.ButtonStyle.success,
+            label="Show all self assignable roles",
+            custom_id=self.CUSTOM_ID,
+            row=1
+        )
+        self.assignable_roles = assignable_roles
+
+    async def callback(self, interaction: Interaction) -> t.Any:
+        """Sends the original subscription view containing the available self assignable roles."""
+        await interaction.response.defer()
+        view = prepare_self_assignable_roles_view(interaction, self.assignable_roles)
+        message = await interaction.followup.send(
```

Comment:
Main reason was my bad understanding of the interaction API, editing that view was failing since it was referencing the wrong message. And I thought I needed to workaround this by keeping a ref to the original message.

Seems like using `interaction.response.edit_message` works well
Fixed in both 7118c51b9b62b2a5ce0e745e4832b17f52901235 and 7976bc2af0fe49ddaebc63242b581b9eecb66132